example/main.go

```diff
@@ -22,9 +21,10 @@ func main() {
 		for {
 			select {
 			case data := <-notificationChannel:
-				fmt.Println(data)
-				bus.UnSubscribe(notification, notificationChannel)
-				waitGroup.Done()
+				if data != nil {
```

Comment:
@arberiii, why do you do this check?

Reply:
Reading from a closed channel returns the zero value (in this case nil). 

Here is how the process goes:
1. Subscribe to the topic
2. Publish message to the topic
3. Read the message
4. At this point, the subscriber wants to unsubscribe, which closes the channel.
5. Inside the infinite loop, it will try to read again from the channel but not it is closed so it returns nil.

I think it is possible to break the for loop after unsubscribing and incrementing the waitGroup.

Reply:
Now, it makes sense. Thank you :)